data/modules/CargoRun/CargoRun.lua

```diff
@@ -367,37 +372,53 @@ local onChat = function (form, ref, option)
 
 	elseif option == 2 then
 		local howmuch
-		if ad.wholesaler then
-			howmuch = string.interp(l["HOWMUCH_WHOLESALER_" .. Engine.rand:Integer(1, getNumberOfFlavours("HOWMUCH_WHOLESALER"))], {
-				amount    = ad.amount,
-				cargoname = ad.cargotype:GetName()})
+		if ad.amount == 1 then
+			howmuch = string.interp(l["HOWMUCH_SINGULAR_" .. Engine.rand:Integer(1,getNumberOfFlavours("HOWMUCH_SINGULAR"))],
+				{amount = ad.amount})
 		else
-			if ad.amount > 1 then
-				howmuch = string.interp(l["HOWMUCH_" .. Engine.rand:Integer(1,getNumberOfFlavours("HOWMUCH"))],
-					{amount = ad.amount})
+			if ad.wholesaler then
+				howmuch = string.interp(l["HOWMUCH_WHOLESALER_" .. Engine.rand:Integer(1, getNumberOfFlavours("HOWMUCH_WHOLESALER"))], {
+					amount    = ad.amount,
+					cargoname = ad.cargotype:GetName()})
 			else
-				howmuch = string.interp(l["HOWMUCH_SINGULAR_" .. Engine.rand:Integer(1,getNumberOfFlavours("HOWMUCH_SINGULAR"))],
+				howmuch = string.interp(l["HOWMUCH_" .. Engine.rand:Integer(1,getNumberOfFlavours("HOWMUCH"))],
 					{amount = ad.amount})
```

Comment:
In general: using the same string twice "HOWMUCH_" and "HOWMUCH", I'm half thinking if this might be made more generic, but I'm not suggesting any change. Just thinking out loud.